pkg/reconciler/domainmapping/resources/ingress.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"knative.dev/networking/pkg/apis/networking"
+	netv1alpha1 "knative.dev/networking/pkg/apis/networking/v1alpha1"
+	"knative.dev/pkg/kmeta"
+	servingv1alpha1 "knative.dev/serving/pkg/apis/serving/v1alpha1"
+)
+
+// MakeIngress creates an Ingress object for a DomainMapping.
+func MakeIngress(dm *servingv1alpha1.DomainMapping, ingressClass string) *netv1alpha1.Ingress {
+	var (
+		targetServiceName      = dm.Spec.Ref.Name
+		targetServiceNamespace = dm.Spec.Ref.Namespace
+	)
+
+	// This assumes the target is a KSvc. To support arbitrary addressables
+	// we will need to resolve the reference to a URL in the same way that
+	// eventing does.
+	targetHostName := targetServiceName + "." + targetServiceNamespace + ".svc.cluster.local"
```

Comment:
This would work with any service that implements our dataplane probe, it just makes an assumption about how URIs are resolved.  I assume you know about the URI resolved in `knative.dev/pkg`?

Reply:
Wasn't suggesting for this PR btw, just hoping to save you time if you didn't.

Reply:
I didn't but I assumed I'd find something along those lines when the time came :D

Reply:
You referring to https://github.com/knative/pkg/blob/master/network/domain.go ?

Reply:
Actually that's a good point, this should use that.

I'm talking about this: https://github.com/knative/pkg/blob/c0496d5e1b2bed5a988c80451061ed5b116057f8/resolver/addressable_resolver.go#L41

Reply:
Yah, TIL you can change cluster.local - cool! Will fix that tmr

Reply:
🤔  if we can do this with KinD, then we probably should try varying that across our KinD matrix 🤩

Reply:
https://github.com/knative/serving/pull/9745